internal/middleware/middleware.go

```diff
@@ -0,0 +1,60 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+)
+
+type Logger struct {
+	Entry *logrus.Logger
+}
+
+func NewLogger() *Logger {
+	return &Logger{Entry: logrus.New()}
+}
+
+func (l Logger) AccessLogMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		next.ServeHTTP(w, r)
+		l.Entry.WithFields(logrus.Fields{
+			"type":        "access log",
+			"method":      r.Method,
+			"remote_addr": r.RemoteAddr,
+			"host":        r.Host,
+		}).Info()
+	})
+}
+
+func (l Logger) TimeLogMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		start := time.Now()
+		next.ServeHTTP(w, r)
+		duration := time.Now().Sub(start)
+		l.Entry.WithFields(logrus.Fields{
+			"type":      "time log",
+			"work_time": duration,
+		}).Info()
+	})
+}
+
+type correlationKey string
```

Comment:
What's the purpose of the new type if you anyway do a type assertion to string?
here: https://github.com/VTerenya/employees/pull/4/files#diff-0097c8e59074fdb549fd2466fb349b2814b208df005f9ccf97b14568d51a5f0aR83